File: AcademicWebApp/src/main/java/com/example/AcademicWebApp/Models/Course.java
package com.example.AcademicWebApp.Models;


import lombok.*;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;

@Data
@Entity(name = "course")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Course {

    @Id
    @Column(name="cid")
    private int cid;

    @Column(name="name")
    private String name;

    @Column(name="fid")
    private int fid;

    @Column(name="year")
    private int year;

    @Column(name="teacher")
    private String teacher;

    @Column(name="semester")
    private int semester;

    @Column(name="maxstudents")
    private int maxstudents;

    @Column(name= "priority")
    private int priority;

    @Column(name= "credits")
    private int credits;

}

File: AcademicApp/src/app/entities/course.ts
export class Course{
  
  constructor(
    public cid: number,
    public name: string,
    public fid: number,
    public year: number,
    public teacher: string,
    public semester: number,
    public maxstudents: number,
    public priority: number,
    public credits: number
   ) {}
}
